### PR TITLE
Add pre index views command

### PR DESCRIPTION
**Pull request by dianabarsan**
Adds a command installs and indexes views from a requested 4.x release. The design documents _ids and versions are changed. 
A bunch of this code is copied from API. Which is unfortunate.

### CODE REVIEW

File: src/pre-index-views.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+const indexView = async (dbName, ddocId, viewName) => {
+  do {
+    try {
+      const url = utils.getUrl(`/${dbName}/${ddocId}/_view/${viewName}`, false, `limit=0`);
+      return await utils.request({ url });
+    } catch (requestError) {
+      if (!requestError || !requestError.response || !TIMEOUT_ERROR.includes(await requestError.response.error)) {
+        throw requestError;
+      }
```
Comment (garethbowen) [inline src/pre-index-views.js]: It's not obvious how this loop continues. Maybe pull out a variable for `timeoutError` to make this self documenting?

Also, do you really need to `await` the `error` access??

```suggestion
      const isTimeout = requestError && requestError.response && TIMEOUT_ERROR.includes(await requestError.response.error);
      if (isTimeout) {
        continue; // views still indexing - try again
      }
      throw requestError;
```

**Review by garethbowen — APPROVED** [review]
Awesome! Shame about the dupe code, but it's probably not worth linking for what should be a temporary script.

One optional readability suggestion inline.

File: src/pre-index-views.js (hunk 1 above)
Comment (dianabarsan) [reply src/pre-index-views.js]: > Also, do you really need to await the error access??

Without the error title, and just the status code, it's sometimes hard to make out what the error is. For example, a CouchDb timeout error is a 500, you only know it's a timeout if you inspect the body. 
I found it really useful with request-promise-native handed you the body of the errored response.

Comment (dianabarsan) [reply src/pre-index-views.js]: On second thought, it seems like I had changed the error handling code in request to wait for the error content ! thanks for flagging this!

**Pull request merged**